samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,275 @@
+// </copyright>
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Teams;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Bot.Connector.Authentication;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using AdaptiveCards.Templating;
+using AdaptiveCards;
+using BotSsoAdaptivecard.Helper;
+
+namespace Microsoft.BotBuilderSamples
+{
+    // This IBot implementation can run any type of Dialog. The use of type parameterization is to allows multiple different bots
+    // to be run at different endpoints within the same project. This can be achieved by defining distinct Controller types
+    // each with dependency on distinct IBot types, this way ASP Dependency Injection can glue everything together without ambiguity.
+    // The ConversationState is used by the Dialog system. The UserState isn't, however, it might have been used in a Dialog implementation,
+    // and the requirement is that all BotState objects are saved at the end of a turn.
+    public class DialogBot<T> : TeamsActivityHandler where T : Dialog
+    {
+        protected readonly BotState _conversationState;
+        protected readonly Dialog _dialog;
+        protected readonly ILogger _logger;
+        protected readonly BotState _userState;
+        protected string _connectionName { get; }
+
+        public DialogBot(ConversationState conversationState, UserState userState, T dialog, ILogger<DialogBot<T>> logger, string connectionName)
+        {
+            _conversationState = conversationState;
+            _userState = userState;
+            _dialog = dialog;
+            _logger = logger;
+            _connectionName = connectionName;
+        }
+
+        /// <summary>
+        /// Get sign in link
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task<string> GetSignInLinkAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var userTokenClient = turnContext.TurnState.Get<UserTokenClient>();
+            var resource = await userTokenClient.GetSignInResourceAsync(_connectionName, turnContext.Activity as Activity, null, cancellationToken).ConfigureAwait(false);
+            return resource.SignInLink;
+        }
+
+        /// <summary>
+        /// Add logic to apply after the type-specific logic after the call to the base class method.
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// The context object for this turn.
+        /// <param name="cancellationToken"></param>
+        /// A cancellation token that can be used by other objects or threads to receive notice of cancellation.
+        /// <returns></returns>
```

Comment:
No explanation added for these parameters

Reply:
Fixed